test/e2e/trafficcontrol_test.go

```diff
@@ -0,0 +1,310 @@
+	if err != nil {
+		t.Fatalf("Error when waiting for IP for Pod '%s': %v", podName, err)
+	}
+	podIP := podIPs.ipv4.String()
+	return podIP
+}
+
+func createPacketsCollectorPod(t *testing.T, data *TestData, podName string) string {
+	require.NoError(t, data.createPodOnNode(podName, data.testNamespace, testNode, agnhostImage, []string{"sleep"}, []string{"3600"}, nil, nil, false, func(pod *corev1.Pod) {
+		privileged := true
+		pod.Spec.Containers[0].SecurityContext = &corev1.SecurityContext{Privileged: &privileged}
+	}))
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, podName, data.testNamespace))
+	podIPs, err := data.podWaitForIPs(defaultTimeout, podName, data.testNamespace)
+	if err != nil {
+		t.Fatalf("Error when waiting for IP for Pod '%s': %v", podName, err)
+	}
+	podIP := podIPs.ipv4.String()
+	return podIP
+}
+
+func (data *TestData) createTrafficControl(t *testing.T,
+	generateName string,
+	matchExpressions []metav1.LabelSelectorRequirement,
+	matchLabels map[string]string,
+	direction v1alpha2.Direction,
+	action v1alpha2.TrafficControlAction,
+	targetPort interface{},
+	isTargetPortVXLAN bool,
+	returnPort interface{}) *v1alpha2.TrafficControl {
+	tc := &v1alpha2.TrafficControl{
+		ObjectMeta: metav1.ObjectMeta{GenerateName: generateName},
+		Spec: v1alpha2.TrafficControlSpec{
+			AppliedTo: v1alpha2.AppliedTo{
+				PodSelector: &metav1.LabelSelector{
+					MatchExpressions: matchExpressions,
+					MatchLabels:      matchLabels,
+				},
+			},
+			Direction:  direction,
+			Action:     action,
+			ReturnPort: &v1alpha2.TrafficControlPort{},
+		},
+	}
+	switch targetPort.(type) {
+	case *v1alpha2.OVSInternalPort:
+		tc.Spec.TargetPort.OVSInternal = targetPort.(*v1alpha2.OVSInternalPort)
+	case *v1alpha2.NetworkDevice:
+		tc.Spec.TargetPort.Device = targetPort.(*v1alpha2.NetworkDevice)
+	case *v1alpha2.UDPTunnel:
+		if isTargetPortVXLAN {
+			tc.Spec.TargetPort.VXLAN = targetPort.(*v1alpha2.UDPTunnel)
+		} else {
+			tc.Spec.TargetPort.GENEVE = targetPort.(*v1alpha2.UDPTunnel)
+		}
+	case *v1alpha2.GRETunnel:
+		tc.Spec.TargetPort.GRE = targetPort.(*v1alpha2.GRETunnel)
+	case *v1alpha2.ERSPANTunnel:
+		tc.Spec.TargetPort.ERSPAN = targetPort.(*v1alpha2.ERSPANTunnel)
+	}
+
+	switch returnPort.(type) {
+	case *v1alpha2.OVSInternalPort:
+		tc.Spec.ReturnPort.OVSInternal = returnPort.(*v1alpha2.OVSInternalPort)
+	case *v1alpha2.NetworkDevice:
+		tc.Spec.ReturnPort.Device = returnPort.(*v1alpha2.NetworkDevice)
+	default:
+		tc.Spec.ReturnPort = nil
+	}
+
+	tc, err := data.crdClient.CrdV1alpha2().TrafficControls().Create(context.TODO(), tc, metav1.CreateOptions{})
+	require.NoError(t, err, "Failed to create TrafficControl")
+	return tc
+}
+
+func getOFPort(t *testing.T, data *TestData, portName string) int {
+	targetOFPort := -1
+	cmd := []string{"ovs-ofctl", "show", defaultBridgeName}
+	if err := wait.Poll(time.Second, 20*time.Second, func() (bool, error) {
```

Comment:
It shouldn't take 20 seconds? I would say at most 3 seconds, otherwise there is efficiency issue in the implementation. The interval could be smaller in case of it's ready between 2~3s.